library/core/src/iter/adapters/mod.rs

```diff
@@ -1687,6 +1687,109 @@ impl<I: Iterator> Peekable<I> {
     {
         self.next_if(|next| next == expected)
     }
+
+    /// Creates an iterator that consumes elements until predicate is
+    /// true, without consuming that last element.
```

Comment:
Not super happy with "that last element". "without consuming the matching element." maybe.

Reply:
Agree but maybe keep the "last" part:

```suggestion
    /// true, without consuming the last matching element.
```

Reply:
I don't want to bikeshed to long, but I'd suggest a comma between last and matching `without consuming the last, matching element`? Otherwise it may sound like multiple elements can match.